### ISSUE TITLE
cfg files from plugins are touched when you update the server

### ISSUE THREAD

**Issue by CJNiiva** [lenosisnickerboa/csgosl#307]
https://youtu.be/rRxfe5tfDQU

TLDW: when updating the server it also adds the basic cfg files back in

Possible solution: add a button in csgosl that makes it so that csgosl stops touching the cfg files, especially for plugin cfg's

**Comment by lenosisnickerboa** [comment]
Thanks for the excellent problem description :)

It's a totally fair point and I would like to support this somehow. Problem is how to do this in a robust way for all cfg files.

Your proposed solution may sound simple but it isn't. Since when updating csgosl the csgosl.zip may contain updated plugins as well and the csgosl install archive contains default cfgs which may be required for the new plugin to run. So, to be on the safe side, during a csgosl update all cfgs are simply overwritten. This would also fix the problem with someone having messed up some cfg causing csgo or the plugin to fail to load.

One obvious way would be to simply add an option where you can disable the "plugin update" when you press "Update server". This way you could update the server without loosing your cfgs. This would be a simple fix.

BUT... that would mean you have to manually enable this option in case you want to update your plugins (as part of updating the server). But if this new option is disabled by default and you enable it, I guess you need to remember this and enable it if needed.

And of course, you would still loose your config when updating csgosl.

So, do you see a point with the option to disable "plugin update" during "Update server"?

**Comment by CJNiiva** [comment]
Yeah, I do see the point, could be an advanced option to just, I dont know, just leave the config, the solution I found was to basically just do what csgosl does in the first place, just take the file structure of the server, with "server
/csgo/cfg" etc to just copy the cfgs I dont want csgosl to change and just copy over my server folder (with just the cfgs in the right place) and overwrite the server to undo the overwrite csgosl made. One solution would be to make a cfg tab and make you manually choose what cfg files to copy before an update, then make csgosl paste them after it has done its usual patching and overwriting? I dont know how to code at all, but it wasnt that hard to do manually, either way man, love the program:) fuck that got convoluted I'm just gonna make a quick showcase vid: https://youtu.be/nYA4k_b5L4o

**Comment by lenosisnickerboa** [comment]
Considered "future enhancement" for now. No time to fix this in a good way :(
